trie/stacktrie.go

```diff
@@ -242,26 +245,26 @@ func (st *StackTrie) getDiffIndex(key []byte) int {
 }
 
 // Helper function to that inserts a (key, value) pair into
-// the trie.
-func (st *StackTrie) insert(key, value []byte) {
+// the trie. Adding the prefix when inserting too.
+func (st *StackTrie) insert(key, value []byte, prefix []byte) {
```

Comment:
So as I understand here, the interface for inserting (key, value) pairs requires also the path. So the `prefix` here is the path from root to the current node, and it is used for `hash()` where we call function to insert the (key, value) pair. Is it correct?

Reply:
and it is used for hash() where we call function to insert the (key, value) pair. Is it correct?-> It will use when u wanna flush to disk when u wanna insert and hash
-> It is the path from root to the current node, when start insert, path is nil 
```	
if st.writeFn != nil {
		st.writeFn(st.owner, path, common.BytesToHash(st.val), h.tmp)
	}
```

Reply:
One example, how it store in. state/snapshot 
```

	var nodeWriter trie.NodeWriteFunc
	// Implement nodeWriter in case db is existed otherwise let it be nil.
	if db != nil {
		nodeWriter = func(owner common.Hash, path []byte, hash common.Hash, blob []byte) {
			scheme.WriteTrieNode(db, owner, path, hash, blob)
		}
	}
	t := trie.NewStackTrieWithOwner(nodeWriter, owner)
```

Reply:
Ok I got it.